### PR TITLE
Apply Job types on Pipeline page(#2eh4wwh)

### PR DESCRIPTION
**Pull request by shashwatbangar**
### Related Issues

Closes #

### Screenshots of Visual Changes before/after (If There Are Any)

**IMPORTANT NOTICE** - Remember to add changelog entry

### Contribution and Currently Important Rules Acceptance

- [ ] I read and followed [contribution rules](https://github.com/hotwax/job-manager#contribution-guideline)

### CODE REVIEW

File: src/types/Job.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+export interface Job {
+    id: string;
+    name: string;
+    systemJobEnum?: Array<Enumeration>;
+    parentJobId?: string;
+    runTime?: string;
+    serviceName?: string;
+    status?: Array<Status>;
+    tempExpr?: Array<TemporalExpression>;
```
Comment (ymaheshwari1) [inline src/types/Job.ts:13]:
```suggestion
    tempExpr?: TemporalExpression;
```

File: src/types/Job.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { Status } from "./Status";
+import { TemporalExpression } from "./temporalExpression";
+import { Enumeration } from "./Enumeration";
+
+export interface Job {
+    id: string;
+    name: string;
+    systemJobEnum?: Array<Enumeration>;
```
Comment (ymaheshwari1) [inline src/types/Job.ts:8]:
```suggestion
    systemJobEnum?: Enumeration;
```

File: src/types/Job.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    parentJobId?: string;
+    runTime?: string;
+    serviceName?: string;
+    status?: Array<Status>;
```
Comment (ymaheshwari1) [inline src/types/Job.ts:12]:
```suggestion
    status?: Status;
```

File: src/types/index.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import { RunningJobsTypes, PendingJobsTypes, JobsHistoryTypes } from "./Job";
+
+export {
+    PendingJobsTypes,
```
Comment (ymaheshwari1) [inline src/types/index.ts:4]: Job.ts file does not contain the imported job types.

Export the other types, Enumeration, TemporalExpression and Status

File: src/views/Pipeline.vue [hunk 5]
```diff
@@ -392,8 +393,10 @@ export default defineComponent({
       return
     },
     async copyJobInformation(job: any) {
+      console.log(job);
```
Comment (ymaheshwari1) [inline src/views/Pipeline.vue:396]: Unwanted console log statement.

File: src/store/modules/job/actions.ts [hunk 6]
```diff
@@ -228,10 +282,37 @@ const actions: ActionTree<JobState, RootState> = {
               'status': job?.statusId
             }
           })
+  
           if(payload.viewIndex && payload.viewIndex > 0){
             jobs = state.pending.list.concat(resp.data.docs);
           }
-          commit(types.JOB_PENDING_UPDATED, { jobs, total });
+          
+          const typeDefinedJobs: Job = jobs.map((response: any) => {
```
Comment (ymaheshwari1) [inline src/store/modules/job/actions.ts:290]:
```suggestion
          const typeDefinedJobs: Job = jobs.map((job: any) => {
```

File: src/store/modules/job/actions.ts [hunk 7]
```diff
@@ -228,10 +282,37 @@ const actions: ActionTree<JobState, RootState> = {
...
+            return { 
+              id: response.jobId,
+              name: response.jobName,
+              systemJobEnum: [
```
Comment (ymaheshwari1) [inline src/store/modules/job/actions.ts:294]: Update this property to an object which will contain key-value pair for id, name, description and typeId

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/views/Pipeline.vue (hunk 5 above)
Comment (shashwatbangar) [reply src/views/Pipeline.vue:396]: Fixed it sir!

File: src/store/modules/job/actions.ts (hunk 6 above)
Comment (shashwatbangar) [reply src/store/modules/job/actions.ts:293]: done sir!

File: src/store/modules/job/actions.ts (hunk 7 above)
Comment (shashwatbangar) [reply src/store/modules/job/actions.ts]: okay sir!

File: src/types/index.ts (hunk 4 above)
Comment (shashwatbangar) [reply src/types/index.ts:4]: done sir!

File: src/types/Job.ts (hunk 3 above)
Comment (shashwatbangar) [reply src/types/Job.ts]: fixed it sir.

File: src/types/Job.ts (hunk 2 above)
Comment (shashwatbangar) [reply src/types/Job.ts]: okay sir done!

File: src/types/Job.ts (hunk 1 above)
Comment (shashwatbangar) [reply src/types/Job.ts]: yes sir!

File: src/types/Job.ts [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+    id: string;
+    name: string;
+    systemJobEnum?: Enumeration;
+    parentJobId?: string;
+    runTime?: string;
+    serviceName?: string;
+    status?: Status;
+    tempExpr?: TemporalExpression;
+    currentRetryCount?: number;
+    finishDateTime: number;
```
Comment (ymaheshwari1) [inline src/types/Job.ts:15]:
```suggestion
    finishDateTime?: number;
```

File: src/views/Pipeline.vue [hunk 9]
```diff
@@ -38,15 +38,15 @@
           </div>
 
           <div v-else>
-            <ion-card v-for="job in pendingJobs" :key="job.jobId" @click="viewJobConfiguration(job)" :button="isDesktop">
+            <ion-card v-for="job in pendingJobs" :key="job.id" @click="viewJobConfiguration(job)" :button="isDesktop">
               <ion-card-header>
-                <ion-card-title>{{ getEnumName(job.systemJobEnumId) }}</ion-card-title>
+                <ion-card-title>{{ getEnumName(job.systemJobEnum.id) }}</ion-card-title>
```
Comment (ymaheshwari1) [inline src/views/Pipeline.vue:43]:
```suggestion
                <ion-card-title>{{ job.systemJobEnum.name }}</ion-card-title>
```

File: src/views/Pipeline.vue [hunk 10]
```diff
@@ -38,15 +38,15 @@
...
                 <ion-badge v-if="job.runTime" color="dark">{{ timeFromNow(job.runTime)}}</ion-badge>
               </ion-card-header>
 
               <ion-item lines="none">
                 <ion-label class="ion-text-wrap">
-                  <p>{{ getEnumDescription(job.systemJobEnumId) }}</p>
+                  <p>{{ getEnumDescription(job.systemJobEnum.id) }}</p>
```
Comment (ymaheshwari1) [inline src/views/Pipeline.vue:49]:
```suggestion
                  <p>{{ job.systemJobEnum.description) }}</p>
```

File: src/views/Pipeline.vue [hunk 11]
```diff
@@ -56,7 +56,7 @@
 
               <ion-item>
                 <ion-icon slot="start" :icon="timerOutline" />
-                <ion-label class="ion-text-wrap">{{ job.tempExprId ? temporalExpr(job.tempExprId)?.description : "🙃"  }}</ion-label>
+                <ion-label class="ion-text-wrap">{{ job.tempExpr.id ? temporalExpr(job.tempExpr.id)?.description : "🙃"  }}</ion-label>
```
Comment (ymaheshwari1) [inline src/views/Pipeline.vue:59]:
```suggestion
                <ion-label class="ion-text-wrap">{{ job.tempExpr.description ? job.tempExpr.description : "🙃"  }}</ion-label>
```

File: src/views/Pipeline.vue [hunk 12]
```diff
@@ -393,7 +394,7 @@ export default defineComponent({
     },
     async copyJobInformation(job: any) {
```
Comment (ymaheshwari1) [inline src/views/Pipeline.vue:395]:
```suggestion
    async copyJobInformation(job:Job) {
```

File: src/views/Pipeline.vue [hunk 13]
```diff
@@ -342,7 +343,7 @@ export default defineComponent({
       isJobDetailAnimationCompleted: false,
       isDesktop: isPlatform('desktop'),
       isRetrying: false,
-      queryString: '' as any
+      queryString: '' as any,
```
Comment (ymaheshwari1) [inline src/views/Pipeline.vue:346]: unwanted change

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/store/modules/job/actions.ts [hunk 14]
```diff
@@ -86,8 +87,36 @@ const actions: ActionTree<JobState, RootState> = {
           }
           jobs.map((job: any) => {
             job['statusDesc'] = this.state.util.statusDesc[job.statusId];
-          })          
-          commit(types.JOB_HISTORY_UPDATED, { jobs, total });
+          }) 
+
+          const typeDefinedJobs: Job = jobs.map((job: any) => {
+            return {
```
Comment (ymaheshwari1) [inline src/store/modules/job/actions.ts:93]: As we are returning an object we don't need a return statement here.

File: src/types/Job.ts (hunk 8 above)
Comment (shashwatbangar) [reply src/types/Job.ts:15]: done sir

File: src/types/Job.ts [hunk 15]
```diff
@@ -0,0 +1,17 @@
...
+    cancelDateTime: number;
```
[the remark this answers is not in the document]
Comment (shashwatbangar) [reply src/types/Job.ts:16]: fixed it!

File: src/views/Pipeline.vue (hunk 9 above)
Comment (shashwatbangar) [reply src/views/Pipeline.vue:43]: done sir!

File: src/views/Pipeline.vue (hunk 11 above)
Comment (shashwatbangar) [reply src/views/Pipeline.vue:59]: ok sir

File: src/views/Pipeline.vue (hunk 10 above)
Comment (shashwatbangar) [reply src/views/Pipeline.vue:49]: sure sir. Fixed it

File: src/views/Pipeline.vue [hunk 16]
```diff
@@ -168,34 +168,34 @@
...
-                <ion-badge v-if="job.statusId" :color="job.statusId === 'SERVICE_FINISHED' ? 'success' : 'danger'">{{ job.statusDesc }}</ion-badge>
+                <ion-badge v-if="job.cancelDateTime || job.finishDateTime" color="dark">{{ job.status.id == "SERVICE_CANCELLED" || job.status.id == "SERVICE_CRASHED" ?  timeFromNow(job.cancelDateTime) : timeFromNow(job.finishDateTime) }}</ion-badge>
+                <ion-badge v-if="job.status.id" :color="job.status.id === 'SERVICE_FINISHED' ? 'success' : 'danger'">{{ job.status.desc }}</ion-badge>
               </div>
             </ion-card-header>
 
             <ion-item lines="none">
               <ion-label class="ion-text-wrap">
-                <p>{{ getEnumDescription(job.systemJobEnumId) }}</p>
+                <p>{{ getEnumDescription(job.systemJobEnum.id) }}</p>
```
[the remark this answers is not in the document]
Comment (shashwatbangar) [reply src/views/Pipeline.vue:185]: sure!

File: src/views/Pipeline.vue (hunk 12 above)
Comment (shashwatbangar) [reply src/views/Pipeline.vue]: sure sir

File: src/views/Pipeline.vue (hunk 13 above)
Comment (shashwatbangar) [reply src/views/Pipeline.vue]: fixed it.

File: src/store/modules/job/actions.ts (hunk 14 above)
Comment (shashwatbangar) [reply src/store/modules/job/actions.ts]: yes sir. Fixed it.

**Comment by adityasharma7** [comment]
Will be done differently. Closing

**Pull request closed without merging**